source/uwp/Renderer/lib/AsyncOperations.h

```diff
@@ -67,7 +77,15 @@ class RenderAsyncBase :
             MakeAgileDispatcherCallback([this]() -> HRESULT
         {
             m_builder->AddListener(this);
-            m_builder->BuildXamlTreeFromAdaptiveCard(m_card.Get(), &m_rootXamlElement, m_renderer.Get());
+            try
+            {
+                m_builder->BuildXamlTreeFromAdaptiveCard(m_card.Get(), &m_rootXamlElement, m_renderer.Get());
+            }
+            catch (...)
+            {
+                //Catch all non-Image loading related problems.
+                return XamlBuilderHadError();
+            }
```

Comment:
Perhaps xamlbuilder should return an HRESULT rather than convert all the hresults to exceptions via THROW_IF_FAILED.  Something to consider when we rework to return a "render result"

Reply:
Yep, definitely. This is not the ideal solution, but since we need to do a more extensive rework in this same area, this is meant as a small patch to mitigate the issues we are already seeing.